woocommerce/woocommerce-gutenberg-products-block
# Add "Filter Products by Stock" block

This PR introduces a new block: "Filter Products by Stock". When this is used in conjunction with the `All Products` block, shoppers will be able to filter out products by different stock levels. This block supports custom stock levels too.


Fixes #4096 




#### Accessibility




### Screenshots

![image](https://user-images.githubusercontent.com/10712881/116401771-ada97480-a83c-11eb-84c0-a67a6562b7ee.png)
![image](https://user-images.githubusercontent.com/10712881/116402010-f2351000-a83c-11eb-9e30-6e4d5d2cfadc.png)





### How to test the changes in this Pull Request:

1. Add 'Filter Products by Stock", "All Products" and "Active Filters" blocks to a page.
2. Ensure `Hide out of stock products from catalogue` is **disabled** in **WooCommerce > Settings > Products > Inventory**
2. Set the stock status of some products to `out of stock`, some to `on backorder`.
3. Go to the page on the front end and verify that you can filter products by the three different stock levels.
4. For any product that is `out of stock` change it back to `in stock`. All products should now either be `on backorder` or `in stock`.
5. Reload the page and ensure the option to filter out of stock products is not there.
5. Set some products back to being `out of stock`.
6.  Change the `Hide out of stock products from catalogue` to **enabled** in **WooCommerce > Settings > Products > Inventory**
7. Reload the page and ensure the option to filter out of stock products is not there.
8. Add the `Filter Products by Price` and `Filter Products by Attribute` block to the page. 
9. Verify that the filters work together and the correct products are shown in the `All Products` block.




### Changelog

>Added "Filter Products by Stock" block
